huaweicloud/services/waf/data_source_huaweicloud_waf_reference_tables.go

```diff
@@ -70,28 +72,31 @@ func DataSourceWafReferenceTablesV1() *schema.Resource {
 	}
 }
 
-func dataSourceWafReferenceTablesRead(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	client, err := config.WafV1Client(config.GetRegion(d))
+func dataSourceWafReferenceTablesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.WafV1Client(conf.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("error creating HuaweiCloud WAF client: %s", err)
+		return diag.Errorf("error creating WAF client: %s", err)
 	}
 
 	opts := valuelists.ListValueListOpts{
-		EnterpriseProjectId: config.GetEnterpriseProjectID(d),
+		EnterpriseProjectId: conf.GetEnterpriseProjectID(d),
 	}
 	r, err := valuelists.List(client, opts)
 	if err != nil {
-		return common.CheckDeleted(d, err, "Error obtain WAF reference table information")
+		return common.CheckDeletedDiag(d, err, "Error obtain WAF reference table information")
 	}
 
 	if len(r.Items) == 0 {
-		return fmtp.Errorf("Your query returned no results. Please change your search criteria and try again.")
+		return diag.Errorf("Your query returned no results. Please change your search criteria and try again.")
 	}
 	// filter data by name
 	filterData, err := utils.FilterSliceWithField(r.Items, map[string]interface{}{
 		"Name": d.Get("name").(string),
 	})
+	if err != nil {
+		return diag.Errorf("error filterdata: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("error filtering WAF reference tables: %s", err)
```
Make the log information more detailed. Please check the other parts.